renku/core/management/datasets.py

```diff
@@ -835,28 +835,15 @@ def _fetch_lfs_files(repo_path, paths):
         except SubprocessError as e:
             raise errors.GitError(f"Cannot pull LFS objects from server: {e}")
 
-    def dataset_commits(self, dataset, max_results=None):
-        """Gets the newest commit for a dataset or its files.
-
-        Commits are returned sorted from newest to oldest.
-        """
-        # FIXME: Return correct hash/dataset id in https://github.com/SwissDataScienceCenter/renku-python/issues/2210
-        # paths = [(self.path / dataset.path / self.METADATA).resolve()]
-        #
-        # paths.extend(self.path / f.path for f in dataset.files)
-        #
-        # commits = self.repo.iter_commits(paths=paths, max_count=max_results)
-        #
-        # return commits
-        return [self.repo.head.commit]
-
-    def add_dataset_tag(self, dataset, tag, description="", force=False):
+    @staticmethod
+    @inject.autoparams()
+    def add_dataset_tag(
```

Comment:
In the spirit of having less functionality on LocalClient, moving this and the remove_dataset_tag code to a dedicated file in `renku/core/management/dataset` would be nice.

Reply:
I moved these to `renku/core/management/dataset/tag`. I believe our namespaces are getting too long; maybe we can drop the `management` part later.

Reply:
I agree. Though maybe we could also move things up a level and "remove" core, that'd help keeping separate things separate in regards to dependencies.

Also `management` isn't a nice name for it. Clean Architecture calls it `use_cases` but I don't like that, either. At least as a developer I'd think it contains documents and diagrams if I saw a folder with that name. `business_logic` is too long. Maybe `controllers`